### PR TITLE
move USE=ieee-long-double to clang

### PR DESCRIPTION
**Pull request by gyakovlev**
doing it via clang-common uncovered some weird interactions.

### CODE REVIEW

**Comment by gyakovlev** [comment]
fedora (who already migrated) does it the same as in this PR:
https://src.fedoraproject.org/rpms/clang/c/54612cbb684f201ff16961595332405cb5ab50d0?branch=rawhide

File: sys-devel/clang/clang-16.0.0_pre20221217.ebuild [hunk 1]
```diff
@@ -273,6 +273,13 @@ multilib_src_configure() {
 
 		-DPython3_EXECUTABLE="${PYTHON}"
 	)
+
+	if use ppc64 && ! use elibc_musl; then
```
Comment (thesamesam) [inline sys-devel/clang/clang-16.0.0_pre20221217.ebuild:277]: Do we actually need this condition given we have the profiles masking as appropriate?

File: sys-devel/clang/clang-15.0.5.ebuild [hunk 2]
```diff
@@ -16,7 +16,7 @@ HOMEPAGE="https://llvm.org/"
 LICENSE="Apache-2.0-with-LLVM-exceptions UoI-NCSA MIT"
```
Comment (thesamesam) [inline sys-devel/clang/clang-15.0.5.ebuild:16]: Please add a note in the commit message as to why it was wrong/didn't work out.

File: sys-devel/clang/clang-16.0.0_pre20221217.ebuild (hunk 1 above)
Comment (gyakovlev) [reply sys-devel/clang/clang-16.0.0_pre20221217.ebuild:277]: ppc64 condition may go actually, for cross purposes too.

musl has to stay for now IMO, strictly unsupported, so it;d better be noop even if someone does something nasty.

```c
bool ToolChain::defaultToIEEELongDouble() const {
  return PPC_LINUX_DEFAULT_IEEELONGDOUBLE && getTriple().isOSLinux();
}
```
https://github.com/llvm/llvm-project/blob/d22f050e15cb8c941d488d9674329db320ef1783/clang/lib/Driver/ToolChain.cpp#L141-L143

**Review by thesamesam — APPROVED** [review]

**Review by mgorny — APPROVED** [review]
LGTM.

**Pull request closed without merging**